### PR TITLE
Cohort connect with services

### PR DESCRIPTION
**Pull request by korgan00**
## Implementation details

Depends on https://github.com/Qiskit/saiba/pull/352

## How to read this PR

`/account/cohort/057c74fa-2efd-4a49-b245-c81bb44142a4/edit`

### CODE REVIEW

File: components/cohort/Edit.vue [hunk 1]
```diff
@@ -86,36 +105,53 @@ import 'carbon-web-components/es/components/button/index';
 import 'carbon-web-components/es/components/checkbox/index';
 import 'carbon-web-components/es/components/file-uploader/index';
 import IconUpload from '@carbon/icons-vue/es/upload/16';
-import type { Cohort } from '@/types/cohort';
+import IconSend from '@carbon/icons-vue/es/send/16';
+import type { Cohort, Member } from '@/types/cohort';
 
 const props = defineProps<{
   cohort: Cohort;
+  isSending: boolean;
```
Comment (yoshiemuranaka) [inline components/cohort/Edit.vue:113]: Not blocking, should we use `isPending` / `pending` to be consistent with $fetch? https://nuxt.com/docs/api/composables/use-async-data#return-values
I don't feel strongly about this, what do you think?

File: components/cohort/MembersList.vue [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
   <div class="cohort-member-list">
     <div
-      v-for="(member, index) in members"
+      v-for="email in invitations"
+      :key="email"
+      class="cohort-member-list__item"
+    >
+      <span class="cohort-member-list__item__invitation">(invitation)</span>
+      <div class="cohort-member-list__item__content">{{ email }}</div>
+      <button
```
Comment (yoshiemuranaka) [inline components/cohort/MembersList.vue:10]: could this be a carbon bx-btn?

**Review by yoshiemuranaka — COMMENTED** [review]
Nice progres 👏 
A few questions after testing the UI flow...

1. Should non-admin users be able to create/edit cohorts? I was able to create new cohorts, but when I tried to edit the `Accelerator-Test` cohort, got a 403
    1. If non-admin users are able to see the cohorts but not perform certain actions, should we disable the actions that are not allowed?
2. We probably need design input, but after deleting a user from the list, it wasn't very obvious to me that the `Update` button still needed to be clicked.
3. I was able to create a cohort with an empty name, should we disallow?



https://user-images.githubusercontent.com/8097306/206308001-7a4c3c63-1100-4e82-9c89-9199ece299f1.mov

File: components/cohort/Edit.vue [hunk 3]
```diff
@@ -75,6 +75,25 @@
...
+      <bx-btn size="md" kind="ghost" @click="handleCancelRequest">
+        Cancel
+      </bx-btn>
+      <bx-btn
+        size="md"
+        icon-layout
+        :disabled="isSending"
+        @click="handleSendRequest"
+      >
+        <slot name="send-text">Update</slot>
```
[the remark this answers is not in the document]
Comment (korgan00) [reply components/cohort/Edit.vue:88]: If you want to fill a slot in web components, you have to use the `slot` attribute. This is from the `Edit.vue` component to change the text from the parent. It has a default content.

File: components/cohort/Edit.vue (hunk 1 above)
Comment (korgan00) [reply components/cohort/Edit.vue:113]: It is ok with me. I don't have a strong opinion about this.

File: components/cohort/Edit.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (yoshiemuranaka) [reply components/cohort/Edit.vue:94]: Oo interesting. Was there a reason you chose having the named slot over a prop like `isEditing`? just curious, not suggesting it should be done that way

**Pull request merged**